### PR TITLE
SPV-918: Destinations Page User flow

### PR DESCRIPTION
**Pull request by Nazarii-4chain**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet-admin/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

### CODE REVIEW

File: src/components/AddDestinationDialog/AddDestinationDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+      toast.error('Failed to add destination');
+    }
+  };
+  const handleMetadataChange = (event: React.ChangeEvent<HTMLTextAreaElement>) => {
+    setMetadata(event.target.value);
+  };
+
+  const handleDialogOpen = () => {
+    setIsAddDialogOpen((prev) => !prev);
+  };
```
Comment (chris-4chain) [inline src/components/AddDestinationDialog/AddDestinationDialog.tsx:66]: The name of the function doesn't fit with the implementation.
Change the name to sth like: `handleDialogToggle` or change the implementation to `setIsAddDialogOpen(true)`.

File: src/routes/login.tsx [hunk 2]
```diff
@@ -38,7 +38,9 @@ export const Route = createFileRoute('/login')({
 
 export function LoginForm() {
   const [role, setRole] = useState<Role>(Role.User);
-  const [key, setKey] = useState('');
+  const [key, setKey] = useState(
+    'xprv9s21ZrQH143K2pDFBrdSqGY3cM47tWJjSC4jWoGivdcjnboKaXohiD9PR5kyMZZApZBZUn9PV4a2Ye6oXsWenHwDbBQKdiauPujcRxLz4U4',
+  );
```
Comment (chris-4chain) [inline src/routes/login.tsx:43]: I think it's not supposed to be here 🙈

File: src/routes/user/_user.destinations.tsx [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+      });
+    }
+  }, [tab]);
+
+  useEffect(() => {
+    navigate({
+      search: (old) => {
+        return {
+          ...old,
+          lockingScript: filter.startsWith('76') ? filter : undefined,
```
Comment (chris-4chain) [inline src/routes/user/_user.destinations.tsx:103]: What do the `startWith` and the `76` mean? 
Maybe you could make a utility function with proper name and comment?

File: src/routes/user/_user.destinations.tsx [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+          address: filter.length > 0 && filter.length <= 34 ? filter : undefined,
```
Comment (chris-4chain) [inline src/routes/user/_user.destinations.tsx:104]: Maybe you could make a constant for that `34` number.

File: src/routes/user/_user.destinations.tsx [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+  }, [debouncedFilter]);
+
+  useEffect(() => {
+    setFilter(lockingScript || address || '');
+    navigate({
+      search: (old) => {
+        return {
+          ...old,
+          lockingScript: lockingScript || filter.startsWith('76') ? filter : undefined,
+          address: address || (filter.length > 0 && filter.length <= 34) ? filter : undefined,
```
Comment (chris-4chain) [inline src/routes/user/_user.destinations.tsx:118]: Same concern here.

File: src/components/AddDestinationDialog/AddDestinationDialog.tsx (hunk 1 above)
Comment (Nazarii-4chain) [reply src/components/AddDestinationDialog/AddDestinationDialog.tsx:66]: Changed

File: src/routes/login.tsx (hunk 2 above)
Comment (Nazarii-4chain) [reply src/routes/login.tsx:43]: Removed

File: src/routes/user/_user.destinations.tsx (hunk 3 above)
Comment (Nazarii-4chain) [reply src/routes/user/_user.destinations.tsx:103]: Created a utility function to determine if string is a lockingScript.

File: src/routes/user/_user.destinations.tsx (hunk 4 above)
Comment (Nazarii-4chain) [reply src/routes/user/_user.destinations.tsx:104]: Created a utility function to determine if string is an address

File: src/routes/user/_user.destinations.tsx (hunk 5 above)
Comment (Nazarii-4chain) [reply src/routes/user/_user.destinations.tsx:118]: Added functions

**Review by chris-4chain — APPROVED** [review]

**Pull request merged**